src/linker/Linker/LinkContext.cs

```diff
@@ -531,6 +537,12 @@ public void LogWarning (string text, int code, MessageOrigin origin, string subc
 			if (!LogMessages)
 				return;
 
+			if ((GeneralWarnAsError && !WarnAsError.Contains (((uint) code, false))) ||
+				(!GeneralWarnAsError && WarnAsError.Contains (((uint) code, true)))) {
```

Comment:
My understanding of the desired behavior is that command line like this:
```console
--warnaserror IL2020 --warnaserror- IL2020
```

Will effectively mean a no-op, that is `IL2020` will be reported as a warning. Currently the code will report this as an error. Basically this part of the "Spec":
> Options are processed in order, last wins

We should also have test for these cases.

I would implement this by changing the `HashSet` to a `Dictionary<uint, bool>`, but there are probably other ways to do this.